build-logic/build-platform/build.gradle.kts

```diff
@@ -58,7 +58,7 @@ dependencies {
         api("com.beust:jcommander:1.78")
         api("org.codehaus.groovy:$groovyVersion")
         api("org.codehaus.groovy.modules.http-builder:http-builder:0.7.2")
-        api("org.codenarc:CodeNarc:2.0.0")
+        api("org.codenarc:CodeNarc:3.0.1")
```

Comment:
If you upgrade the CodeNarc version here you need to capture that in the potentially breaking changes/release notes since this changes the default version for CodeNarc the Gradle codenarc plugin uses.

Reply:
You're mostly right. I think this is the one we ship, but the default is actually set here:

https://github.com/gradle/gradle/blob/abffe87cdcd417bc1f462e19d8a0b35cd2891134/subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CodeNarcPlugin.java#L42

I'll just revert this for this PR.

Reply:
Ah, this is `build-logic:build-platform`, so this is not something we ship, but something we use in our build. So nothing to update here.

Reply:
That'd be distribution-dependencies, right?

Reply:
And if we define the version here, why do we have to explicitly state the version in the other two places, too? Couldn't we omit it there?